src/main/java/seedu/rc4hdb/logic/commands/modelcommands/FilterCommand.java

```diff
@@ -34,29 +35,30 @@ public class FilterCommand implements ModelCommand {
             + "[" + PREFIX_HOUSE + "HOUSE] "
             + "[" + PREFIX_MATRIC_NUMBER + "MATRIC_NUMBER] "
             + "[" + PREFIX_TAG + "TAG]...\n"
-            + "Example: " + COMMAND_WORD + " 1 "
+            + "Example: " + COMMAND_WORD
             + PREFIX_PHONE + "91234567 "
             + PREFIX_EMAIL + "johndoe@example.com";
 
     public static final String MESSAGE_NOT_FILTERED = "At least one field to filter must be provided.";
 
     /** description to filter the resident with */
     private final ResidentDescriptor filterPersonDescriptor;
+    private final FilterSpecifier specifier;
 
     /**
      * @param filterPersonDescriptor description object to filter the resident with
      */
-    public FilterCommand(ResidentDescriptor filterPersonDescriptor) {
+    public FilterCommand(ResidentDescriptor filterPersonDescriptor, FilterSpecifier specifier) {
         requireNonNull(filterPersonDescriptor);
         this.filterPersonDescriptor = new ResidentDescriptor(filterPersonDescriptor);
+        this.specifier = specifier;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        //return new CommandResult("Command Still in progress");
         requireNonNull(model);
         AttributesMatchKeywordsPredicate predicate =
-                new AttributesMatchKeywordsPredicate(filterPersonDescriptor);
+                new AttributesMatchKeywordsPredicate(filterPersonDescriptor, specifier);
```

Comment:
You could consider splitting this up into 2 different predicates. Would decrease coupling as each predicate will specialize in their own specific `/all` and `/any` filtering.

Reply:
Perhaps we could do a `AttributesMatchAllKeywordsPredicate` and `AttributesMatchAnyKeywordsPredicate`.